internal_types/src/entry.rs

```diff
@@ -0,0 +1,1544 @@
+
+    Ok(entry_fb::PartitionWrite::create(
+        fbb,
+        &entry_fb::PartitionWriteArgs {
+            key: Some(partition_key),
+            table_batches: Some(table_batches),
+        },
+    ))
+}
+
+fn build_table_write_batch<'a>(
+    fbb: &mut FlatBufferBuilder<'a>,
+    table_name: &str,
+    lines: Vec<&'a ParsedLine<'_>>,
+    default_time: &DateTime<Utc>,
+) -> Result<flatbuffers::WIPOffset<entry_fb::TableWriteBatch<'a>>> {
+    let mut columns = BTreeMap::new();
+    for (i, line) in lines.iter().enumerate() {
+        let row_number = i + 1;
+
+        if let Some(tagset) = &line.series.tag_set {
+            for (key, value) in tagset {
+                let key = key.as_str();
+                let builder = columns
+                    .entry(key)
+                    .or_insert_with(ColumnBuilder::new_tag_column);
+                builder.null_to_row(row_number);
+                if builder.push_tag(value.as_str()).is_err() {
+                    return TableColumnTypeMismatch {
+                        table: table_name,
+                        column: key,
+                    }
+                    .fail();
+                }
+            }
+        }
+
+        for (key, val) in &line.field_set {
+            let key = key.as_str();
+
+            match val {
+                FieldValue::Boolean(b) => {
+                    let builder = columns
+                        .entry(key)
+                        .or_insert_with(ColumnBuilder::new_bool_column);
+                    builder.null_to_row(row_number);
+                    if builder.push_bool(*b).is_err() {
+                        return TableColumnTypeMismatch {
+                            table: table_name,
+                            column: key,
+                        }
+                        .fail();
+                    }
+                }
+                FieldValue::U64(v) => {
+                    let builder = columns
+                        .entry(key)
+                        .or_insert_with(ColumnBuilder::new_u64_column);
+                    builder.null_to_row(row_number);
+                    if builder.push_u64(*v).is_err() {
+                        return TableColumnTypeMismatch {
+                            table: table_name,
+                            column: key,
+                        }
+                        .fail();
+                    }
+                }
+                FieldValue::F64(v) => {
+                    let builder = columns
+                        .entry(key)
+                        .or_insert_with(ColumnBuilder::new_f64_column);
+                    builder.null_to_row(row_number);
+                    if builder.push_f64(*v).is_err() {
+                        return TableColumnTypeMismatch {
+                            table: table_name,
+                            column: key,
+                        }
+                        .fail();
+                    }
```

Comment:
I wonder if you could add a `source: Box<Error>` field to `TableColumnTypeMismatch` and then write this code like:

```suggestion
                    builder.push_f64(*v)
                      .map_err(|e| Box::new(e) as _)
                      .context(TableColumnTypeMismatch {
                            table: table_name,
                            column: key,
                        })?
```

But now that I wrote that I am not sure how much better that really is -- this works too

Reply:
Found a decent way to represent this: a7659addbe8362aae6ceff16bbaddde7ab9e7e6e